### ISSUE TITLE
Sortable columns on the Coverage and Alleles tables

### ISSUE THREAD

**Issue by lchorbadjiev** [iossifovlab/gain#984]
## Parent

iossifovlab/gain#982

## What to build

Make the Coverage and Alleles per-chromosome tables on a genomic score's
info page sortable by every column, by clicking the column header.

These tables run to hundreds of rows on a human genome — the Coverage
table on a hg38 score has one row per contig — and there is currently no
way to answer "which contig has the worst coverage?" other than reading
all of them.

Sorting is client-side over the already-rendered rows: these are static
generated pages with no server behind them.

### The opt-in contract

Two attributes, and a table that carries neither is untouched:

- `<th data-sort="text">` / `<th data-sort="number">` — the header
  declares that it sorts and how to compare.
- `<td data-sort-value="…">` — the cell carries the key. An absent
  attribute means "no value" and sorts last in **both** directions.

This keeps display formatting out of the comparator. `Covered %` renders
`90.45%` but carries `data-sort-value="0.9045"`; a row whose `fraction`
is `None` renders an empty cell and carries no attribute at all, rather
than an empty string that `parseFloat` would turn into `NaN` and scatter
unpredictably. Chromosome carries the natural-order key from
iossifovlab/gain#983.

Deliberately **not** sortable: the substitution matrix (its row headers
are `<th>`, not data) and the four-row Allele classes table.

### Table structure

All three per-chromosome tables — Coverage, Alleles and **Fragments** —
get an explicit `<tbody>` for data rows and a `<tfoot>` holding the
`all chromosomes` totals row, so the sorter reorders `tbody > tr` and
physically cannot displace the total.

Fragments gets the structure but **not** the sort attributes, so that
opting it in later is two attributes and nothing else. Leaving one of
three sibling tables in the old shape is how this kind of thing drifts.

This also removes an existing inconsistency: the totals row is marked
`class="coverage-global"` in Coverage and Fragments but is unmarked in
Alleles, and that class has no CSS behind it anywhere. Delete it —
`tfoot` is the selector now. Explicit `<tbody>` is already the house
style (`grr_index.jinja`, the Files table in `resource_template.jinja`,
`statistics_template.jinja`); these three tables are the outliers.

### The sorter

A new `sortable_table.jinja` partial, `{% include %}`d from
`resource_template.jinja`, in **vanilla JS**.

Do not reuse the sorter in `grr_scripts.jinja`: it is jQuery-shaped, and
that include pulls jQuery 3.7.1 from `ajax.googleapis.com`. Resource
pages are jQuery-free today, and dragging in a CDN dependency and a third
external origin to do one `Array.sort()` is a bad trade. The hard parts —
parsing and ordering — already live in Python via `data-sort-value`.

Placing it in `resource_template.jinja` means it ships on resource pages
that have no sortable table (gene score, gene models, reference genome).
That is fine: it is inert without opt-in markup, and it sits with the
modal JS already there.

### Header affordance

Two-state cycle: ascending, descending, ascending. No third "unsorted"
state — the original order is an artifact of the statistics file's
insertion order, so a way back to it is a way back to noise. (This
differs from `grr_index.jinja`, where the unsorted state is the
repository's own meaningful listing order.)

Every sortable header shows an indicator at all times — dimmed when idle,
full strength on the active column — so the click target is discoverable
on an otherwise entirely static page, and the sorted column is
identifiable at a glance rather than by reading arrow directions.

Use Material Symbols (`unfold_more` / `arrow_upward` / `arrow_downward`)
to match `grr_index.jinja`, but **subset the font URL** with
`&icon_names=arrow_upward,arrow_downward,unfold_more`. Resource pages do
not currently load an icon font at all; `grr_index.jinja` pulls the full
variable icon font with `&display=block`, which is heavy for three
glyphs. Retrofitting the subset onto `grr_index.jinja` is a cheap
follow-up, not part of this.

## Acceptance criteria

- [ ] Coverage and Alleles tables sort by every column, ascending and
      descending, on header click
- [ ] Coverage, Alleles and Fragments each have an explicit `<tbody>`,
      with the `all chromosomes` row in a `<tfoot>`
- [ ] The totals row never moves, under any sort, on any column
- [ ] `.coverage-global` is gone from templates
- [ ] Fragments has the structure but no sort attributes
- [ ] Numeric columns sort numerically, not lexicographically (`9` before
      `10`, not after)
- [ ] `Covered %` sorts by the underlying fraction; a row with no
      fraction renders an empty cell, carries no `data-sort-value`, and
      sorts last both ascending and descending
- [ ] The substitution matrix and Allele classes tables carry no sort
      attributes and are unaffected
- [ ] No jQuery on the page; no new external origin beyond the subset
      Material Symbols stylesheet
- [ ] Ties preserve the natural chromosome order from
      iossifovlab/gain#983 (`Array.prototype.sort` is stable as of
      ES2019, so this follows from the initial DOM order)
- [ ] Rendered-source tests in `core/tests/small/templates/` assert the
      markup contract: `tbody`/`tfoot` placement, `data-sort` on the
      right `<th>`s, `data-sort-value` on the `<td>`s, and **neither** on
      the two opted-out tables

## Notes

Behavioural coverage of the click itself is out of scope and is tracked
separately — CI has no JS runtime. See the follow-up issue.

Existing tests assert cell contents as `f">{COVERED}<"`, which anchors on
`>value<` and survives adding attributes to the `<td>` opening tag.

## Blocked by

None — iossifovlab/gain#983 merged (PR iossifovlab/gain#997), so the
natural chromosome key this issue emits as `data-sort-value` now exists.

Two things about it differ from what #983's body sketched, and both
matter here:

- It lives at `core/gain/utils/chromosome_order.py`, **not** under
  `genomic_resources/statistics/` — placed there precisely so this
  issue can import it from the template layer, which imports nothing
  from `genomic_resources`.
- The key is `natural_chromosome_key(chrom) -> str`, and its encoding
  is *not* the `chr|0000000010|_gl|…` form #983 sketched. That form
  fails #983's own criterion (`|` is `0x7C`, `m` is `0x6D`, so
  `chr|0000000022|` sorts after `chrm`). What shipped: each digit run
  becomes its digits behind a two-digit count of them, no separator.
  Do not reconstruct the key in JS — read it off the attribute.

One caveat for the comparator: the key is deliberately non-injective
(leading zeros are stripped, so `chr01` and `chr1` collide). Ties must
fall through to the sort's stability, which this issue's acceptance
criteria already assume.

**Comment by lchorbadjiev** [comment]
> *This was generated by AI during triage.*

## Triage Notes

**State:** `ready-for-agent`, retaining `blocked` until #983 lands. The body is
accepted as the agent brief.

**Verified before labelling — every structural claim is accurate:**

- The totals rows really are inconsistent: `class="coverage-global"` is present
  on the Coverage and Fragments totals rows and **absent** on the Alleles one.
  An implementer who greps for `.coverage-global` to find "the totals rows"
  will find two of three; the body already calls this out, which is why it is
  worth keeping in the brief rather than trusting the grep.
- That class has no CSS rule behind it anywhere in the repo, so deleting it is
  inert.
- None of the three tables has an explicit `<tbody>` today — rows sit directly
  under `<table>` after `</thead>`. The sorter must therefore be scoped to
  `tbody > tr`; an `Array.from(table.querySelectorAll('tr'))` would sweep the
  header row in with the data.
- `grr_scripts.jinja` is included only by `grr_index.jinja`, so the "do not
  reuse the jQuery sorter / stay jQuery-free" constraint costs nothing.

**Ordering dependency is real, not bookkeeping.** #983 produces the string key
this issue emits as `data-sort-value` on the Chromosome column, and the
stable-sort tie-breaking criterion depends on #983 having already fixed the
initial DOM order. The `blocked` label is correct.

**Comment by lchorbadjiev** [comment]
> *This was generated by AI during triage.*

## Triage Notes — resumed, unblocked

**State:** `ready-for-agent`, `blocked` removed. iossifovlab/gain#983 closed as
completed today, merged as PR iossifovlab/gain#997 (`a1d578cc7` on `master`).
This issue is now genuinely pickable.

### The dependency landed better than the brief assumed

`natural_chromosome_key` (`core/gain/utils/chromosome_order.py`) returns a plain
string orderable by `<` alone — no tuple, no comparator — explicitly so it can be
emitted as a template sort attribute. Ties are left to a stable sorter, which is
what this issue's tie-breaking criterion relies on.

### Re-verified after iossifovlab/gain#997 — every structural claim in the body still holds

iossifovlab/gain#997 sorted **in Python only** (`sorted(..., key=natural_chromosome_key)` in
`statistics/coverage.py` and `statistics/alleles.py`). It did not touch the
markup. So, still true on `master`:

- None of the three tables has an explicit `<tbody>`; rows sit directly under
  `<table>` after `</thead>`.
- `class="coverage-global"` is still on the Coverage totals row
  (`genomic_score.jinja:127`) and the Fragments one (`fragment_score.jinja:21`),
  and still absent from Alleles. Still no CSS behind it anywhere.

One scope note: iossifovlab/gain#997 ordered **Fragments** too, not just Coverage and Alleles.
That does not change this issue's plan — Fragments still gets structure without
sort attributes — but the Fragments table is no longer untouched territory.

### Two refinements for the implementer

**1. The key is not reachable from a template today.** The brief says
"Chromosome carries the natural-order key from iossifovlab/gain#983", which reads as though the
key is already in the markup. It is not — it is consumed inside the statistics
modules and thrown away.

Plumbing it through the row objects does not work uniformly: Coverage and
Fragments render `CoverageRow` / `FragmentRow` NamedTuples
(`statistics/coverage.py:690,745`), but **Alleles has no row type at all** — the
template iterates the plain `dict[str, AlleleCounts]` returned by
`AlleleStatistics.by_chromosome()`.

The seam iossifovlab/gain#983 intended is the Jinja environment. Its module docstring says the
helper lives under `gain/utils/` rather than `genomic_resources/statistics/`
precisely because "iossifovlab/gain#984 wants this key at the template layer, which imports
nothing from `genomic_resources` today". Register it beside the existing global
at `core/gain/templates/__init__.py:133` (`env.globals["markdown"]`) and call it
from all three templates. That is house style and it works for Alleles.

**2. The Notes' reassurance about existing tests is now wrong for the test that
matters most.** The body says existing tests anchor on `>value<` and survive
attributes added to the `<td>` opening tag. True of the older assertions — but
iossifovlab/gain#983 added `core/tests/small/genomic_resources/test_info_page_chromosome_order.py`,
whose row extractor is:

```python
_ROW = re.compile(r"<tr[^>]*>\s*<td>([^<]*)</td>")
```

It requires a **bare** `<td>`. Once the Chromosome cell carries
`data-sort-value`, it matches nothing — verified: the same input returns
`['chr1']` today and `[]` with the attribute present, so all three parametrized
cases of `test_per_chromosome_rows_render_in_natural_order` fail. Updating that
regex is part of this issue, not a surprise for review.

That file is also the natural place to extend for the markup-contract criterion —
it already builds position, allele and fragment resources and renders full pages.
The body's `core/tests/small/templates/` is right for a
`natural_chromosome_key`-is-registered test (alongside `test_markdown_global.py`),
but the rendered-markup assertions belong next to iossifovlab/gain#983's.

### Related, not blocking

iossifovlab/gain#992 proposes consolidating the near-identical `<h2>`-section splitters across the
statistics tests, one of which is the helper above. iossifovlab/gain#987 remains the follow-up
holding behavioural coverage of the click itself.
